packages/cli/src/amplify.ts

```
@@ -2,10 +2,36 @@
 
 import { hideBin } from 'yargs/helpers';
 import { createMainParser } from './main_parser_factory.js';
-import { attachUnhandledExceptionListeners } from './error_handler.js';
+import {
+  attachUnhandledExceptionListeners,
+  extractSubCommands,
+} from './error_handler.js';
+import {
+  AmplifyFault,
+  PackageJsonReader,
+  UsageDataEmitterFactory,
+} from '@aws-amplify/platform-core';
+import { fileURLToPath } from 'node:url';
 
-attachUnhandledExceptionListeners();
+const packageJson = new PackageJsonReader().read(
+  fileURLToPath(new URL('../package.json', import.meta.url))
+);
+const libraryVersion = packageJson.version;
 
-const parser = createMainParser();
+if (libraryVersion == undefined) {
+  throw new AmplifyFault('UnknownVersionFault', {
+    message:
+      'Library version cannot be determined. Check the library installation',
+  });
+}
 
+const usageDataEmitter = await new UsageDataEmitterFactory().getInstance(
+  libraryVersion
+);
+
+attachUnhandledExceptionListeners(usageDataEmitter);
+
+const parser = createMainParser(libraryVersion, usageDataEmitter);
 await parser.parseAsync(hideBin(process.argv));
+
+await usageDataEmitter.emitSuccess({}, { command: extractSubCommands(parser) });
```

This should be in a try catch with any exception thrown swallowed.